### PR TITLE
test: add plugins support to e2e test

### PR DESCRIPTION
**Pull request by alf-devnull**
### What was the problem/requirement? (What/Why)

Automated Project Plugins collecting as Job dependencies should be covered with e2e tests

### What was the solution? (How)

Implement e2e test that check all edge cases:
1. Include Project/Engine plugins, enable and disable them
2. Launch e2e tests
3. After job is finished, parse output and check that expected Plugins were initialized in Unreal Engine session

### What is the impact of this change?

e2e for plugins support implemented

### How was this change tested?

e2e test

### Have you run a render job successfully with these changes?

Yes

### Was this change documented?

Yes

### Is this a breaking change?

No
----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: test/end_to_end/conftest.py [hunk 1]
```diff
@@ -1699,3 +1764,89 @@ def reusable_queue_fleet_association(
...
+    Args:
+        logs_client: The CloudWatch Logs client
+        log_group: The name of the log group
+        log_stream: The name of the log stream
+
+    Returns:
+        A sorted list of unique project plugin names
+    """
+    plugin_names = set()
+    pattern = re.compile(r"Mounting Project plugin\s+(?P<name>.+?)\s*$")
```
Comment (Cherie-Chen) [inline test/end_to_end/conftest.py]: NIT: Please address SonarQube feed back here.

I think `Mounting Project plugin\s+(?P<name>[^\s]+(?:\s+[^\s]+)*)\s*$` should fix it.

File: test/end_to_end/conftest.py [hunk 2]
```diff
@@ -1699,3 +1764,89 @@ def reusable_queue_fleet_association(
...
+            log_response["options"]["logStreamName"],
+        )
+        return plugin_names
+
+    except Exception as e:
+        logger.warning(f"Error fetching job logs or extracting plugins: {str(e)}")
+        return []
+
+
+def _extract_project_plugins_from_log_events(logs_client, log_group, log_stream):
```
Comment (Cherie-Chen) [inline test/end_to_end/conftest.py]: NIT: please add return data type

File: test/end_to_end/test_worker_agent.py [hunk 3]
```diff
@@ -45,6 +49,22 @@ def test_create_job_with_worker_agent(
...
+        worker_project_plugins = get_last_session_project_plugins(
+            deadline_client=deadline_client,
+            farm_id=farm_id,
+            queue_id=queue_id,
+            job_id=job_id,
+        )
+
+        # The worker should have loaded EmptyContentPlugin2 and EmptyContentPlugin3 only
+        assert "EmptyContentPlugin1" not in worker_project_plugins
+        assert "EmptyContentPlugin2" in worker_project_plugins
```
Comment (Cherie-Chen) [inline test/end_to_end/test_worker_agent.py:116]: Running E2E test failed 

```
======================================================== short test summary info =========================================================
FAILED test/end_to_end/test_worker_agent.py::test_create_job_with_worker_agent - AssertionError: assert 'EmptyContentPlugin2' in []
================================================ 1 failed, 1 passed in 696.41s (0:11:36) =================================================
```

File: test/end_to_end/conftest.py [hunk 4]
```diff
@@ -767,7 +825,14 @@ def create_readonly_test_project(request) -> Generator[Tuple[str, str], None, No
 
     # Add our plugins
     project_path = os.path.join(dest_path, "TP_DMXBP.uproject")
-    add_plugins_to_project(project_path, ["UnrealDeadlineCloudService", "MovieRenderPipeline"])
+    add_plugins_to_project(
+        project_path,
+        ["UnrealDeadlineCloudService", "MovieRenderPipeline", "EmptyContentPlugin2"],
+        True,
+    )
+    add_plugins_to_project(project_path, ["EmptyContentPlugin1"], False)
```
Comment (baxeaz) [inline test/end_to_end/conftest.py]: I think it might make sense to test these as part of a new fixture or helper function rather than add them here where they'll be part of  anything reusing the readonly_test_project fixture, and only really verified in the test_create_job_with_worker_agent test.  Also it seems like we're expecting to verify some particular cases here since we're adding some plugins as disabled plugins, some as enabled plugins, and some as enabled content plugins, and then some in multiple cases.  To me those are things that would be good to call out in our test what we're adding and why

**Comment by Cherie-Chen** [comment]
Formatting issue detected in conftest.py by code scanning tool.

To resolve:
1. View issues: Run `hatch run lint`
2. Auto-fix formatting: Run `hatch run fmt`

File: test/end_to_end/conftest.py [hunk 5]
```diff
@@ -767,10 +825,29 @@ def create_readonly_test_project(request) -> Generator[Tuple[str, str], None, No
...
+    add_plugins_to_project(
+        project_path,
+        ["UnrealDeadlineCloudService", "MovieRenderPipeline"],
+        True,
+    )
 
     yield dest_path, project_path
 
+@pytest.fixture(scope="session")
+def create_test_project_with_content_plugins(create_readonly_test_project) -> Generator[Tuple[str, str], None, None]:
```
Comment (Cherie-Chen) [inline test/end_to_end/conftest.py]: I recommend moving the test case setup directly into the test file rather than keeping it in the fixture. This would improve readability and make the test's intent more obvious. Currently, it's unclear which plugins are configured and their enabled/disabled states. By including this setup within the test file itself, it will be more transparent what conditions we're actually testing.

Please also take @baxeaz's comment above in adding a new test for this test case

File: test/end_to_end/conftest.py
Comment (Cherie-Chen) [inline test/end_to_end/conftest.py:1]: I am getting this warming on line 815:
```
============================================================ warnings summary ============================================================
test\end_to_end\conftest.py:815
  C:\Users\Administrator\deadline-cloud-for-unreal-engine\test\end_to_end\conftest.py:815: SyntaxWarning: invalid escape sequence '\T'
    source_path = os.path.join(engine_root, "Templates\TP_DMXBP")
```

The issue occurs because Python interprets '\T' as an invalid escape sequence. Could you please update it to the following?
```
source_path = os.path.join(engine_root, r"Templates\TP_DMXBP")
```

**Comment by Cherie-Chen** [comment]
With this change, developers running E2E test need to grant `GetLogEvents` permissions to their test role. Could you please add a note in [DEVELOPMENT.md](https://github.com/aws-deadline/deadline-cloud-for-unreal-engine/blob/mainline/DEVELOPMENT.md) so new developers won't need to spent time debugging this permission issue.

**Comment by Cherie-Chen** [comment]
All 3 tests are passing

```
================================================================================= slowest 5 durations =================================================================================
449.58s call     test/end_to_end/test_worker_agent.py::test_worker_agent_project_plugins
437.38s call     test/end_to_end/test_worker_agent.py::test_create_job_with_worker_agent
201.47s setup    test/end_to_end/test_create_job.py::test_create_job
85.25s call     test/end_to_end/test_create_job.py::test_create_job
10.13s setup    test/end_to_end/test_worker_agent.py::test_create_job_with_worker_agent
=========================================================================== 3 passed in 1193.88s (0:19:53) ============================================================================
```

**Review by Cherie-Chen — APPROVED** [review]

**Review by baxeaz — APPROVED** [review]
